x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -207,65 +211,75 @@ async function claimAvailableTasks(opts: TaskClaimerOpts): Promise<ClaimOwnershi
   }
 
   // perform the task object updates, deal with errors
-  const updatedTasks: ConcreteTaskInstance[] = [];
+  const updatedTaskIds: string[] = [];
   let conflicts = staleTasks.length;
   let bulkUpdateErrors = 0;
   let bulkGetErrors = 0;
 
-  const updateResults = await taskStore.bulkUpdate(taskUpdates, {
-    validate: false,
-    excludeLargeFields: true,
-  });
+  const updateResults = await taskStore.bulkPartialUpdate(taskUpdates);
   for (const updateResult of updateResults) {
     if (isOk(updateResult)) {
-      updatedTasks.push(updateResult.value);
+      updatedTaskIds.push(updateResult.value.id);
     } else {
-      const { id, type, error } = updateResult.error;
-      if (error.statusCode === 409) {
+      const { id, type, error, status } = updateResult.error;
+
+      // check for 409 conflict errors
+      if (status === 409) {
         conflicts++;
       } else {
-        logger.error(`Error updating task ${id}:${type} during claim: ${error.message}`, logMeta);
+        logger.error(
+          `Error updating task ${id}:${type} during claim: ${JSON.stringify(error)}`,
+          logMeta
+        );
         bulkUpdateErrors++;
       }
     }
   }
 
   // perform an mget to get the full task instance for claiming
-  const fullTasksToRun = (await taskStore.bulkGet(updatedTasks.map((task) => task.id))).reduce<
-    ConcreteTaskInstance[]
-  >((acc, task) => {
-    if (isOk(task)) {
-      acc.push(task.value);
-    } else {
-      const { id, type, error } = task.error;
-      logger.error(`Error getting full task ${id}:${type} during claim: ${error.message}`, logMeta);
-      bulkGetErrors++;
-    }
-    return acc;
-  }, []);
+  const fullTasksToRun = (await taskStore.bulkGet(updatedTaskIds)).reduce<ConcreteTaskInstance[]>(
+    (acc, task) => {
+      if (isOk(task)) {
+        acc.push(task.value);
+      } else {
+        const { id, type, error } = task.error;
+        logger.error(
+          `Error getting full task ${id}:${type} during claim: ${error.message}`,
+          logMeta
+        );
+        bulkGetErrors++;
+      }
+      return acc;
+    },
+    []
+  );
 
   // separate update for removed tasks; shouldn't happen often, so unlikely
   // a performance concern, and keeps the rest of the logic simpler
   let removedCount = 0;
   if (removedTasks.length > 0) {
     const tasksToRemove = Array.from(removedTasks);
+    const tasksToRemoveUpdates: PartialConcreteTaskInstance[] = [];
     for (const task of tasksToRemove) {
-      task.status = TaskStatus.Unrecognized;
+      tasksToRemoveUpdates.push({
+        id: task.id,
+        version: task.version,
```

Comment:
I think we probably have the seqNum/primaryTerm for these, in `docLatestVersions`, however I'm guessing you used the `version` to save from having to create another extremely similar type :-)

I'm wondering, for the removal of unrecognized tasks, maybe we don't even need the version?  These docs shouldn't be updated by anyone anyway, except this code to update their `status`.

Reply:
Makes sense to me. Updated in https://github.com/elastic/kibana/pull/191760/commits/e05a9772372b75095a272baf2cb1687a7f7dcd84